internal/services/firewall/firewall_policy_resource.go

```diff
@@ -142,6 +143,163 @@ func resourceFirewallPolicy() *pluginsdk.Resource {
+											string(network.FirewallPolicyIntrusionDetectionStateTypeDeny),
+										}, false),
+										Optional: true,
+									},
+									"id": {
+										Type:     pluginsdk.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"traffic_bypass": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"name": {
+										Type:     pluginsdk.TypeString,
+										Required: true,
+									},
+									"description": {
+										Type:     pluginsdk.TypeString,
+										Optional: true,
+									},
+									"protocol": {
+										Type:     pluginsdk.TypeString,
+										Required: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											string(network.FirewallPolicyIntrusionDetectionProtocolICMP),
+											string(network.FirewallPolicyIntrusionDetectionProtocolANY),
+											string(network.FirewallPolicyIntrusionDetectionProtocolTCP),
+											string(network.FirewallPolicyIntrusionDetectionProtocolUDP),
+										}, false),
+									},
+									"source_addresses": {
+										Type:     pluginsdk.TypeSet,
+										Optional: true,
+										Elem: &pluginsdk.Schema{
+											Type: pluginsdk.TypeString,
+										},
+									},
+									"destination_addresses": {
+										Type:     pluginsdk.TypeSet,
+										Optional: true,
+										Elem: &pluginsdk.Schema{
+											Type: pluginsdk.TypeString,
+										},
+									},
+									"destination_ports": {
+										Type:     pluginsdk.TypeSet,
+										Optional: true,
+										Elem: &pluginsdk.Schema{
+											Type: pluginsdk.TypeString,
+										},
+									},
+									"source_ip_groups": {
+										Type:     pluginsdk.TypeSet,
+										Optional: true,
+										Elem: &pluginsdk.Schema{
+											Type: pluginsdk.TypeString,
+										},
+									},
+									"destination_ip_groups": {
+										Type:     pluginsdk.TypeSet,
+										Optional: true,
+										Elem: &pluginsdk.Schema{
+											Type: pluginsdk.TypeString,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"identity": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"type": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.ResourceIdentityTypeNone),
+								string(network.ResourceIdentityTypeUserAssigned),
+							}, false),
+						},
+						"principal_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"tenant_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"identity_ids": {
```

Comment:
I think this is better as `user_assigned_identity_ids` so it's clearer that it only applies for that configuration.

```suggestion
						"user_assigned_identity_ids": {
```